src/framework/multiinstances/internal/multiinstancesprovider.cpp

```diff
@@ -210,8 +210,8 @@ bool MultiInstancesProvider::isHasAppInstanceWithoutProject() const
         IF_ASSERT_FAILED(!args.empty()) {
             return false;
         }
-        ret = args.at(0).toInt();
-        if (ret) {
+        if (args.at(0).toInt()) {
+            ret = true;
```

Comment:
existing bug that meant this function didn't reliably return the right result (as one instance might return "true" but this result would be overwritten by the next one that returned "false")

Reply:
I'm wondering why we even capture ret by reference in the lambda. It seems to be an async operation, so capturing local variables by reference is pretty dangerous. @igorkorsukov could you please help with this one?

Reply:
Also, implicitly casing bool to int isn't very good

Reply:
there **sync**RequestToAll - i.e. a synchronous method, to which a callback is passed, which is called for each instance

Reply:
> existing bug that meant this function didn't reliably return the right result (as one instance might return "true" but this result would be overwritten by the next one that returned "false") 

hmm it shouldn't be like that.. because if we return true from this callback, then the poll should be interrupted 
```
Code IpcChannel::syncRequestToAll(const QString& method, const QStringList& args, const OnReceived& onReceived)
{
...

        bool success = onReceived(msg.args);
        if (success) {
            loop.exit(Code::Success);
            return;
        }
```

But it looks like your option is more reliable.

Reply:
You're right, in principle returning true should mean it stops the loop at that point. But I was definitely seeing it not work reliably when there were more than 2 instances until I made this change.  Agree "ret" should be a bool, just didn't want to change any more than necessary.

Reply:
So in fact "syncRequestToAll" does only send a *single* socket message, and in most cases the response is received from all of them very quickly, and the call to loop.exit( ) when one of them returns doesn't stop the onSocketMsgReceived( ) function for the others from calling the callback - in principle we could do that by resetting m_msgCallback to a nullptr.
Also "ret" is an integer because in fact what's passed to onSocketMsgReceived is just a QString, and it has to be interpreted as an int (using toInt()) in order to determine whether it's true or false.

But now I have a bigger problem, and it's an existing problem, which is that I only want to send the message to activate the instance and bring the window to the foreground to a *single* instance, which I haven't figured out how to do yet.

Reply:
Ok, have rewritten as per my suggestion to ensure only a single instance is sent the request to bring its window to the fore (and optionally show the file-new dialog)